File: app/src/main/java/com/example/thinkdo/fragment/DialogFragment.java
package com.example.thinkdo.fragment;

import android.app.AlertDialog;
import android.app.Fragment;
import android.os.Bundle;
import androidx.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ProgressBar;

import com.example.thinkdo.compoentdemo.R;

/**
 * Created by xh on 2018/4/3.
 */

public class DialogFragment extends Fragment {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.frag_dialog, container, false);
        Button btn = (Button) view.findViewById(R.id.btn);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ProgressBar progressBar = new ProgressBar(getActivity(), null, android.R.attr.progressBarStyleLargeInverse);
                ProgressBar progressBar1 = new ProgressBar(getActivity(), null, android.R.attr.progressBarStyleLargeInverse);


                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
//                builder.setTitle("Title");
//                builder.setMessage("content");
//                builder.setCustomTitle(progressBar);
                builder.setCancelable(true);
                builder.setView(progressBar1);
//                builder.setNegativeButton("negative", new DialogInterface.OnClickListener() {
//                    @Override
//                    public void onClick(DialogInterface dialog, int which) {
//
//                    }
//                });
//                builder.setPositiveButton("positiveButton", new DialogInterface.OnClickListener() {
//                    @Override
//                    public void onClick(DialogInterface dialog, int which) {
//
//                    }
//                });
                builder.show();
            }
        });

        return view;
    }
}

File: kotlin/src/main/java/com/example/xh/kotlin/fragment/DialogFragment.kt
package com.example.xh.kotlin.fragment

import android.app.AlertDialog
import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import com.example.xh.kotlin.BaseFragement
import com.example.xh.kotlin.LeakActivity
import com.example.xh.kotlin.R
import kotlinx.android.synthetic.main.fragment_dialog.*

class DialogFragment : BaseFragement() {
    override fun getLayoutResId(): Int {
        return R.layout.fragment_dialog
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val list = arrayOf("苹果", "梨子", "香蕉", "苹果", "梨子", "香蕉", "苹果", "梨子", "香蕉", "苹果", "梨子", "香蕉")

        val booleanArray = booleanArrayOf(false, true, false, false, true, false, false, true, false, false, true, false)
        button.setOnClickListener {
            val itemString: Array<CharSequence> = arrayOf("item one", "item Two")
            AlertDialog.Builder(activity)
                    .setTitle("title")
                    .setItems(itemString) { _: DialogInterface, which: Int ->
                        Toast.makeText(activity, "${itemString[which]}", Toast.LENGTH_SHORT).show()
//                            d.dismiss()
                    }
                    .setMultiChoiceItems(list, booleanArray) { t1, t2, t3 ->
                        Log.d("TAG", "$t2")
                    }
                    .setNeutralButton("选择全部", null)
                    .setPositiveButton("确定", null)
                    .setNegativeButton("取消", null)
                    .show()
        }

        btn2.setOnClickListener {
            startActivity(Intent(activity, LeakActivity::class.java))
        }


    }
}